components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/GeoLocationBasedServiceImpl.java

```diff
@@ -350,6 +472,49 @@ public Response getGeoAlertsHistory(@PathParam("deviceId") String deviceId,
         }
     }
 
+    @Path("alerts/history")
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    public Response getGeoAlertsHistoryForGeoClusters(@QueryParam("from") long from, @QueryParam("to") long to) {
+        String tableName = "IOT_PER_DEVICE_STREAM_GEO_ALERTNOTIFICATIONS";
+        String fromDate = String.valueOf(from);
+        String toDate = String.valueOf(to);
+        String query = "";
+        if (from != 0 || to != 0) {
+            query = "timeStamp : [" + fromDate + " TO " + toDate + "]";
+        }
+        try {
+            List<SortByField> sortByFields = new ArrayList<>();
+            SortByField sortByField = new SortByField("timeStamp", SortType.ASC);
+            sortByFields.add(sortByField);
+
+            // this is the user who initiates the request
+            String authorizedUser = MultitenantUtils.getTenantAwareUsername(
+                    CarbonContext.getThreadLocalCarbonContext().getUsername());
+
+            try {
```

Comment:
Suggestion. Move this nested try block into a separate method or catch the exception under the main try-catch block.

Reply:
Fixed c653051